### DISCUSSION TITLE
Ways to generate contours with other than 5 m distans

### DISCUSSION THREAD

**Discussion by stefankinell** [karttapullautin/karttapullautin#61] [General]
In the past I have been playing around with different methods of generating contours with other interval than 5 meters. If I understand the original script correctly it will as basis always smoothed generate in the vector output
* contours at 5
* intermediate contours at 2,5 m (e.g. between the 5 m contours)
* formlines in seperate file, that overlap with the intermediate contours, which is natural.

When I want to use contours on different height I have sometimes tried the parameter scalefactor, but then it seems formlines are still in original non-scaled 2,5 meters interval. 

More often I instead use las2las and scale the z-factors of the las-file before I run pullauta. But then I often run it twice to get normal cliffs and green/yellows. 

IN one of this years potential project I was looking into generating contours on 2,5 m interval, with formlines inbetween. Which will be very nice in flatter areas of Sweden. 

Has anyone else got other alternative methods you currently use? 

Before starting I was also thinking of looking into the code and try to understand where the 5 meters (and 2,5 for intermediate) is set. I mean, I assume that is just a number and that it would be possible to alternate to something else, hardcoded or even better with a variable?

**Comment by rphlo** [comment]
Have you tried to play with the scalefactor parameter?

**Reply by stefankinell** [reply]
Yes, I have used scale factor. But one thing here is that the formlines do not seem to be affected. They are still on the 2,5 meters level. At least that is what I remember I have found. And I'm not sure but I think I found the contours to be "nicer" when doing transformation of laz-file first and then running pullauta with manipulated laz-file.

I'll see if I can do a rerun of an area and explain with images.

**Reply by rphlo** [reply]
Actually I would try editing line 596 and 598 `2.5 * scalefactor` to `1.25 * scalefactor`.
If this work as intended this `2.5` could be changed into a parameter

**Reply by stefankinell** [reply]
The adaptions you suggested generated nice curves. It seems to do the trick well.

I also had to do some changes to the identifications to the curves, where you name _intermed etc. I will see If I can post a pull request with the changes, and a variable to use instead of hard coding :-) I assume now it is enough to propose only change the main.rs-file, right?

**Reply by rphlo** [reply]
The only perl code still in use is related to drawing shape files. So yes only rust need to be edited.

**Reply by stefankinell** [reply]
Good. Then I know if I in the future come up with new ideas to contribute with.
